dv/verilator/simutil_verilator/cpp/verilator_sim_ctrl.h

```diff
@@ -17,11 +18,26 @@ enum VerilatorSimCtrlFlags {
   ResetPolarityNegative = 1,
 };
 
+// Callback function to be called every clock cycle
+// The parameter is simulation time
+typedef std::function<void(int /* sim time */)> SimCtrlCallBack;
+
 class VerilatorSimCtrl {
  public:
   VerilatorSimCtrl(VerilatedToplevel *top, CData &clk, CData &rst_n,
                    VerilatorSimCtrlFlags flags = Defaults);
 
+  /**
+   * A helper function to execute a standard set of run commands
+   * The return value of this method is the success/fail return code
+   */
+  int StandardRun(int argc, char **argv);
```

Comment:
I'm not really convinced by the name, but I cannot really think of a significantly better one either. (Probably the same problem you had.)

To make things at least a bit better, can you extend the doxygen comment a bit to describe what this method does, potentially with a bit of sample code? Essentially: call this in your main() to create a simulation with a typical set of command line arguments, signal handlers, etc. (Maybe list the things that are done when running this way.)